utils/obsproc/IcecAmsr2Ioda.h

```diff
@@ -0,0 +1,117 @@
+      int dimySize = ncFile.getDim("Number_of_Y_Dimension").getSize();
+      int dimTimeSize = ncFile.getDim("Time_Dimension").getSize();
+      int nobs = dimxSize * dimySize;
+      int ntimes = dimxSize * dimySize * dimTimeSize;
+
+      // Set the int metadata names
+      std::vector<std::string> intMetadataNames = {};
+      // std::vector<std::string> intMetadataNames = {"pass", "cycle", "mission"};
+
+      // Set the float metadata name
+      std::vector<std::string> floatMetadataNames = {};
+      // std::vector<std::string> floatMetadataNames = {"mdt"};
+      // Create instance of iodaVars object
+      gdasapp::IodaVars iodaVars(nobs, floatMetadataNames, intMetadataNames);
+
+      oops::Log::debug() << "--- iodaVars.location: " << iodaVars.location << std::endl;
+
+      // Read non-optional metadata: datetime, longitude and latitude
+      std::vector<float> oneDimLatVal(iodaVars.location);
+      ncFile.getVar("Latitude").getVar(oneDimLatVal.data());
+
+      std::vector<float> oneDimLonVal(iodaVars.location);
+      ncFile.getVar("Longitude").getVar(oneDimLonVal.data());
+
+      // Read optional integer metadata "Flags"
+      std::vector<int64_t> oneDimFlagsVal(iodaVars.location);
+      ncFile.getVar("Flags").getVar(oneDimFlagsVal.data());
+
+      // Get Ice_Concentration obs values and attributes
+      std::vector<int> oneDimObsVal(iodaVars.location);
+      ncFile.getVar("NASA_Team_2_Ice_Concentration").getVar(oneDimObsVal.data());
+
+      // Read and process the dateTime
+      std::vector<int> oneTmpdateTimeVal(ntimes);
+      ncFile.getVar("Scan_Time").getVar(oneTmpdateTimeVal.data());
+      iodaVars.referenceDate = "seconds since 1970-01-01T00:00:00Z";
+
+      std::tm timeinfo = {};
+      for (int i = 0; i < ntimes; i += dimTimeSize) {
+        for (int j = 0; j < dimTimeSize && i + j < ntimes; j++) {
+        }
+        timeinfo.tm_year = oneTmpdateTimeVal[i] - 1900;  // Year since 1900
+        timeinfo.tm_mon = oneTmpdateTimeVal[i + 1] - 1;  // 0-based; 8 represents Sep
+        timeinfo.tm_mday = oneTmpdateTimeVal[i + 2];
+        timeinfo.tm_hour = oneTmpdateTimeVal[i + 3];
+        timeinfo.tm_min = oneTmpdateTimeVal[i + 4];
+        timeinfo.tm_sec = oneTmpdateTimeVal[i + 5];
+
+        // Calculate and store the seconds since the Unix epoch
+        time_t epochtime = std::mktime(&timeinfo);
+        iodaVars.datetime(i/6) = static_cast<int64_t>(epochtime);
+      }
+
+      // Update non-optional Eigen arrays
+      for (int i = 0; i < iodaVars.location; i++) {
+        iodaVars.longitude(i) = oneDimLonVal[i];
+        iodaVars.latitude(i) = oneDimLatVal[i];
+        iodaVars.obsVal(i) = static_cast<int64_t>(oneDimObsVal[i]*0.01);
+        iodaVars.obsError(i) = 0.1;  // Do something for obs error
+        iodaVars.preQc(i) = oneDimFlagsVal[i];
+        // Save QC Flags in optional floatMetadata
+        // iodaVars.intMetadata(i, 0) = oneDimFlagsVal[i];
+      }
```

Comment:
I don't know yet how to use `Eigen::Map` and change the type, but you should be able to use `Map` for `longitude`, `latitude` and `preQC` (I just added an example in develop). There's also a cleaner way to set obserror to 0.1.
I know this is probably what I did in the RADS example ... But **do as @rmclaren says, not as I do!**

Reply:
This is what 'amsr2_icec2ioda.py' has had, Initially, preQc was set to 0 but it was changed as 'amsr2_icec2ioda.py' is written.